File: ch07/VbServerControl1/Listing07-43.vb
Imports System.Web.UI
Imports System.ComponentModel

<ToolboxData("<{0}:Listing0743 runat=server></{0}:Listing0743>")>
Public Class Listing0743
    Inherits System.Web.UI.WebControls.WebControl

    <Bindable(True), Category("Appearance"), DefaultValue(""),
    Editor(
            GetType(System.Web.UI.Design.UrlEditor),
            GetType(System.Drawing.Design.UITypeEditor))>
    Public Property Url() As String

    Protected Overrides Sub RenderContents(ByVal output As HtmlTextWriter)
        output.Write(Url.ToString())
    End Sub
End Class

File: ch07/CSharpServerControl1/Listing07-43.cs
using System;
using System.ComponentModel;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace CSharpServerControl1
{
    [ToolboxData("<{0}:Listing0743 runat=server></{0}:Listing0743>")]
    public class Listing0743 : WebControl
    {
        [Bindable(true)]
        [Category("Appearance")]
        [DefaultValue("")]
        [Editor(typeof(System.Web.UI.Design.UrlEditor),
            typeof(System.Drawing.Design.UITypeEditor))]
        public string Url { get; set; }

        protected override void RenderContents(HtmlTextWriter output)
        {
            output.Write(this.Url);
        }
    }
}